substrate/client/db/src/lib.rs

```diff
@@ -1880,66 +2143,85 @@ impl<Block: BlockT> Backend<Block> {
 		&self,
 		transaction: &mut Transaction<DbHash>,
 		displaced: &DisplacedLeavesAfterFinalization<Block>,
+		clean_body: bool,
+		clean_state: bool,
```

Comment:
probably something I wrote, but having two bool here is a bit hard to read, likely an enum would be nicer.

Reply:
actually two different function would make more senses (I think it is either call for body or state, not both at the same time).